src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Tests.cs

```diff
@@ -167,6 +167,41 @@ public void PriorityQueue_Generic_EnqueueRange_Null()
             Assert.Equal("not null", queue.Dequeue());
         }
 
+        [Fact]
+        public void PriorityQueue_Generic_Remove_MatchingElement()
```

Comment:
will be good to add a test inserting same element more than once but with different priority then remove and validate the first match is always removed.

Reply:
There's no guarantee of what element will get removed in case of duplicates, but you're right that there should some test exercising that particular scenario.